sklearn/cluster/tests/test_dbscan.py

```diff
@@ -108,10 +108,11 @@ def test_dbscan_sparse_precomputed_different_eps():
 
 @pytest.mark.parametrize("use_sparse", [True, False])
 @pytest.mark.parametrize("metric", ["precomputed", "minkowski"])
-def test_dbscan_input_not_modified(use_sparse, metric):
+@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
+def test_dbscan_input_not_modified(use_sparse, metric, csr_container):
     # test that the input is not modified by dbscan
     X = np.random.RandomState(0).rand(10, 10)
-    X = sparse.csr_matrix(X) if use_sparse else X
+    X = csr_container(X) if use_sparse else X
```

Comment:
This could be optimized by not generating the cases 2 meaningless cases for `use_sparse=False, csr_container=csr_matrix` and `use_sparse=False, csr_container=csr_array`.

See #27099 for a similar example to drop `use_sparse` and instead parametrize `csr_container` with `CSR_CONTAINERS + [None]`.